features/FEATURE_LWIP/lwip-interface/LWIPMemoryManager.h

```diff
@@ -0,0 +1,141 @@
+     * @param align    Memory alignment requirement in bytes
+     * @return         Allocated memory buffer, or NULL in case of error
+     */
+    virtual emac_mem_buf_t *alloc_heap(uint32_t size, uint32_t align);
+
+    /**
+     * Allocates memory buffer from a pool
+     *
+     * Memory buffer allocated from pool is contiguous if size is equal or less than
+     * (aligned) allocation unit, otherwise may be chained. Will typically come from
+     * fixed-size packet pool memory.
+     *
+     * @param  size    Size of the memory to allocate in bytes
+     * @param  align   Memory alignment requirement in bytes
+     * @return         Allocated memory buffer, or NULL in case of error
+     */
+    virtual emac_mem_buf_t *alloc_pool(uint32_t size, uint32_t align);
+
+    /**
+     * Get memory buffer pool allocation unit
+     *
+     * Returns the maximum size of contiguous memory that can be allocated from a pool.
+     *
+     * @param align    Memory alignment requirement in bytes
+     * @return         Contiguous memory size
+     */
+    virtual uint32_t get_pool_alloc_unit(uint32_t align);
+
+    /**
+     * Free memory buffer
+     *
+     * If memory buffer is chained must point to the start of the chain. Frees all buffers
+     * from the chained list.
+     *
+     * @param mem      Memory buffer to be freed.
+     */
+    virtual void free(emac_mem_buf_t *buf);
+
+    /**
+     * Return total length of a memory buffer
+     *
+     * Returns a length of this buffer and if the buffer is chained, the total length of
+     * following buffers in the list.
+     *
+     * @param mem      Memory buffer
+     * @return         Total length in bytes
+     */
+    virtual uint32_t get_total_len(emac_mem_buf_t *buf);
+
+    /**
+     * Copy a memory buffer
+     *
+     * Copies a buffer. For chained buffers, both buffers must point to the start of the
+     * chain. Copy operation does not adjust the lengths of the copied to memory buffer,
+     * so buffer total lengths must be the same.
+     *
+     * @param to_buf    Memory buffer to copy to
+     * @param from_buf  Memory buffer to copy from
+     */
+    virtual void copy(emac_mem_buf_t *to_buf, emac_mem_buf_t *from_buf);
+
+    /**
+     * Concatenate two memory buffers
+     *
+     * Concatenates buffer to end of the other buffer. Concatenated to buffer total length
+     * is adjusted accordingly. For chained buffers, both buffers must point to the start
+     * of the chain.
+     *
+     * @param to_buf   Memory buffer to concatenate to
+     * @param cat_buf  Memory buffer to concatenate
+     */
+    virtual void cat(emac_mem_buf_t *to_buf, emac_mem_buf_t *cat_buf);
+
+    /**
+     * Returns the next buffer
+     *
+     * Returns the next buffer from the memory buffer chain.
+     *
+     * @param mem      Memory buffer
+     * @return         The next memory buffer, or NULL if last
+     */
+    virtual emac_mem_buf_t *next(emac_mem_buf_t *buf);
```

Comment:
Again, const, and same for the others. Should this be `get_next` to match the below?